skglm/estimators.py

```diff
@@ -952,6 +957,174 @@ def fit(self, X, y):
         return _glm_fit(X, y, self, Quadratic(), penalty, solver)
 
 
+class SCADRegression(LinearModel, RegressorMixin):
+
+    r"""Linear regression with SCAD penalty estimator.
+
+    The optimization objective for SCADRegression is, with :math:`x >= 0`:
+
+    .. math::
+        "pen"(x) = {
+            (alpha x                  , if \ \ \ \ \ \ \ \ \ \ x <= alpha),
+            (2 alpha gamma x - x^2 - alpha^2 / 2 (gamma - 1)
+                                      , if       alpha \ \   < x <= alpha gamma),
+            (alpha^2 (gamma + 1) / 2  , if       alpha gamma < x  )
+        :}
+
+    .. math::
+        "value" = sum_(j=1)^(n_"features") "pen"(abs(w_j))
+
+    For more details see
+    `Coordinate descent algorithms for nonconvex penalized regression,
+    with applications to biological feature selection, Breheny and Huang
+    <https://doi.org/10.1214/10-aoas388>`_.
+
+    Parameters
+    ----------
+    alpha : float, optional
+        Penalty strength.
+
+    gamma : float, default=3.7
+        If `` gammma = 1 `` this corresponds to hard thresholding
+
+        If `` gamma = np.inf `` his corresponds to soft thresholding
+
+
+    weights : array, shape (n_features,), optional (default=None)
+        Positive weights used in the L1 penalty part of the Lasso
+        objective. If ``None``, weights equal to 1 are used.
+
+    max_iter : int, optional
+        The maximum number of iterations (subproblem definitions).
+
+    max_epochs : int
+        Maximum number of CD epochs on each subproblem.
+
+    p0 : int
+        First working set size.
+
+    verbose : bool or int
+        Amount of verbosity.
+
+    tol : float, optional
+        Stopping criterion for the optimization.
+
+    fit_intercept : bool, optional (default=True)
+        Whether or not to fit an intercept.
+
+    warm_start : bool, optional (default=False)
+        When set to ``True``, reuse the solution of the previous call to fit as
+        initialization, otherwise, just erase the previous solution.
+
+    ws_strategy : str
+        The score used to build the working set. Can be ``fixpoint`` or ``subdiff``.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,)
+        parameter vector (:math:`w` in the cost function formula)
+
+    sparse_coef_ : scipy.sparse matrix, shape (n_features, 1)
+        ``sparse_coef_`` is a readonly property derived from ``coef_``
+
+    intercept_ : float
+        constant term in decision function.
+
+    n_iter_ : int
+        Number of subproblems solved to reach the specified tolerance.
+
+    See Also
+    --------
+    Lasso : Lasso regularization.
+    """
+
+    def __init__(self, alpha=1., gamma=3.7, max_iter=50, max_epochs=50_000,
+                 p0=10, verbose=0, tol=1e-4, fit_intercept=True,
+                 warm_start=False, ws_strategy="subdiff"):
+        super().__init__()
```

Comment:
To check, but I'm not sure this is needed

Reply:
@mathurinm was refering to the ``super().__init__()``.

Let's keep it since
- it is recommended to call it to init any attributes of the class in the inheritance tree
  though here we don't make use of it.
- to stay consistent with the code base